packages/eslint-plugin/src/rules/require-object-type-annotations.ts

```diff
@@ -0,0 +1,55 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import type { RuleListener } from '@typescript-eslint/utils/src/ts-eslint';
+import type * as ts from 'typescript';
+
+import * as util from '../util';
+
+export default util.createRule({
+  name: 'require-object-type-annotations',
+  defaultOptions: [],
+  meta: {
+    docs: {
+      description: '',
+      requiresTypeChecking: true,
+      recommended: false,
+    },
+    messages: {
+      forbidden: 'Object is missing type annotation.',
+    },
+    schema: [],
+    type: 'problem',
+  },
+  create: (context): RuleListener => {
+    const listener = (esNode: TSESTree.ObjectExpression): void => {
+      const parserServices = util.getParserServices(context);
+      const checker = parserServices.program.getTypeChecker();
```

Comment:
nit - move these outside the listener to the `create` body

Reply:
3935c66a9152dd1940f4184264eb15023a08d6e5